### ISSUE TITLE
Not working anymore?

### ISSUE THREAD

**Issue by InsulinJunkieDe** [pazaan/decoding-contour-next-link#14]
Hi,

I tried to get this up & running. But I'm alway stuck at the this place (after making it work on python3, see my pull-request):

```
# Opening device
Manufacturer: Bayer HealthCare LLC
Product: Contour Link USB Device
Serial No: 000000000105XXXX
# Request Device Info
Please plug in your Contour NextLink 2.4, and rerun this script
```
Since I was able to get this this point, I could see that this one should be sending an 
`self.sendMessage( struct.pack( '>B', 0x58 ) )`

to the device but then nothon seems to happen, because It doesn't get back the serial. 
Although it seems that's already given back when opening the device?

After some research I found out that  within sendMessage

```
    def sendMessage( self, payload ):
        # Split the message into 60 byte chunks
        for packet in [ payload[ i: i+60 ] for i in range( 0, len( payload ), 60 ) ]:
            message = struct.pack( '>3sB', 'ABC', len( packet ) ) + packet
            self.device.write( bytearray( message ) )
            #print ("SEND: " + binascii.hexlify( message )) # Debugging
```

no messages are being send out to the stick, when turning on the debug print.

Do you have any thoughts why this could be issue?

Thanks in advance

**Comment by pazaan** [comment]
When the initial send doesn't work, it's possible that the stick is in a weird state (which can happen a bit during development). What happens when you unplug the stick for 10 seconds, plug it in again, and retry?

**Comment by InsulinJunkieDe** [comment]
I just replugged it in after 13 days, but with the same result.
Do you have any other idea?

**Comment by pazaan** [comment]
The only thing I can think of is that the `hid` import isn't working properly for Python 3? I won't have access to a CNL for a few weeks to test with Python 3 (or test and merge your PR). I'll do that once I have the CNL again :)

**Comment by paulokow** [comment]
Fixed to work with Python 3 on bf6cef6389b7cafaa05060d4f261e513e225c63e
@InsulinJunkieDe please test this branch and report any issues
